librz/analysis/var.c

```diff
@@ -1206,45 +1180,43 @@ RZ_API RZ_OWN RzList /*<RzAnalysisVar *>*/ *rz_analysis_var_list(RZ_NONNULL RzAn
 	void **it;
 	rz_pvector_foreach (&fcn->vars, it) {
 		RzAnalysisVar *var = *it;
-		if (var->kind != kind) {
+		if (var->storage.type != kind) {
 			continue;
 		}
 		rz_list_append(list, var);
 	}
 	return list;
 }
 
-static int var_comparator(const RzAnalysisVar *a, const RzAnalysisVar *b) {
-	// avoid NULL dereference
-	return (a && b) ? (a->delta > b->delta) - (a->delta < b->delta) : 0;
+static int stackvar_comparator(const RzAnalysisVar *a, const RzAnalysisVar *b) {
+	if (!a || !b || a->storage.type != RZ_ANALYSIS_VAR_STORAGE_STACK || b->storage.type != RZ_ANALYSIS_VAR_STORAGE_STACK) {
+		return 0;
+	}
+	return a->storage.stack_off - b->storage.stack_off;
 }
 
 static int regvar_comparator(const RzAnalysisVar *a, const RzAnalysisVar *b) {
-	// avoid NULL dereference
 	return (a && b) ? (a->argnum > b->argnum) - (a->argnum < b->argnum) : 0;
 }
 
 RZ_API void rz_analysis_fcn_vars_cache_init(RzAnalysis *analysis, RzAnalysisFcnVarsCache *cache, RzAnalysisFunction *fcn) {
```

Comment:
Doxygen

Reply:
I will deprecate it. This does not make much sense anymore.